src/core/runtime/projection.h

```diff
@@ -38,10 +38,15 @@ class LegateProjectionFunctor : public Legion::ProjectionFunctor {
   virtual bool is_functional(void) const { return true; }
   virtual bool is_exclusive(void) const { return true; }
   virtual unsigned get_depth(void) const { return 0; }
+  virtual bool is_collective() { return _is_collective; }
+  virtual void set_collective() { _is_collective = true; }
 
  public:
   virtual Legion::DomainPoint project_point(const Legion::DomainPoint& point,
                                             const Legion::Domain& launch_domain) const = 0;
+
+ private:
+  bool _is_collective = false;
```

Comment:
a nitpick: suffixing, not prefixing, member variables with `_` has been the convention.